WebUSB: Prevent excessive recreation of libusb context

EchterAgo opened the pull request:
The context is very small and does not need to be recreated each time. Recreation can actually cause a null-dereference in libusb close as destroying the context (which happens on reassignment) closes all its devices, but they are closed again by the transport.

EchterAgo commented:
Note the `USBContext` closes itself on garbage collection.

cculianu commented:
Hey just to be a nitpicker -- I think this is great and definitely is the right change. It's basically how Trezor does it.

Can we -- for Python purity's sake though.. just pre-populate `cls.context = None` at the class level (eg: `context = None` in class level decls) and then have this code read as:
` if not cls.context: #... `?

I hate to be that guy.. but I'm that guy. :)

EchterAgo commented:
Yea, I'll change it. For reference, here is what trezor does: https://github.com/trezor/trezor-firmware/blob/master/python/trezorlib/transport/webusb.py#L110

EchterAgo commented:
Ok, I updated this. I also added the `atexit` call, though in my testing it doesn't make a difference, it cleans up fine either way. Maybe that isn't the case with older `libusb` versions though. Tested on Windows 10 1903 & Arch Linux.

cculianu commented:
Yeah I'm not clear on why that call would be necessary -- presumably upon process exit the file handles are closed by the kernel drivers anyway so.. meh. :)

Cool. 👍

cculianu commented on keepkeylib/transport_webusb.py:
```diff
@@ -77,8 +79,10 @@ def _close(self):
+        if not cls.context:
+            cls.context = usb1.USBContext()
+            cls.context.open()
+            atexit.register(cls.context.close)
```
atexit needs to be imported... `import atexit` .. no?  Why is it this didn't crash? ;)

EchterAgo commented:
Fixed missing `atexit` import.
